File: modules/photo-module/server/entities/BoardAsset.ts
import { Entity, Column, PrimaryColumn, CreateDateColumn, UpdateDateColumn, ManyToOne, JoinColumn } from "typeorm";
import { Board } from "./Board";

@Entity()
export class BoardAsset {
    @PrimaryColumn("uuid")
    board_id!: string;

    @PrimaryColumn("text")
    asset_name!: string;

    @Column("real", { default: 1.0 })
    scale!: number;

    @Column("real", { default: 0.0 })
    rotation!: number;

    @Column("real", { default: 0.0 })
    x_position!: number;

    @Column("real", { default: 0.0 })
    y_position!: number;

    @Column("text")
    src!: string;

    @UpdateDateColumn({ type: "timestamptz" })
    last_update!: Date;

    @ManyToOne(() => Board, { onDelete: 'CASCADE' })
    @JoinColumn({ name: "board_id" })
    board!: Board;
}

File: HomeStock-backend/src/main/kotlin/com/snk/HomeStock/repository/model/BoardAsset.kt
package com.snk.HomeStock.repository.model

import jakarta.persistence.Column
import jakarta.persistence.EmbeddedId
import jakarta.persistence.Entity
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.MapsId
import jakarta.persistence.Table
import org.hibernate.annotations.UpdateTimestamp
import java.time.OffsetDateTime

@Entity
@Table(name = "board_asset")
class BoardAsset(
    @EmbeddedId
    var id: BoardAssetKey,

    @ManyToOne(optional = false)
    @MapsId("boardId")
    @JoinColumn(name = "board_id")
    var board: Board,

    @Column(nullable = false)
    var scale: Float = 1.0f,

    @Column(nullable = false)
    var rotation: Float = 0.0f,

    @Column(name = "x_position", nullable = false)
    var xPosition: Float = 0.0f,

    @Column(name = "y_position", nullable = false)
    var yPosition: Float = 0.0f,

    @Column(nullable = false)
    var src: String,

    @UpdateTimestamp
    @Column(name = "last_update", nullable = false)
    var lastUpdate: OffsetDateTime? = null
)
